src/org/zaproxy/zap/extension/saml/SAMLUtils.java

```diff
@@ -61,24 +61,26 @@ public static String b64Encode(byte[] data) {
      * @throws SAMLException
      */
     public static String inflateMessage(byte[] data) throws SAMLException {
-        try {
-            Inflater inflater = new Inflater(true);
-            inflater.setInput(data);
-            byte[] xmlMessageBytes = new byte[MAX_INFLATED_SIZE];
-            int resultLength = inflater.inflate(xmlMessageBytes);
-
-            if (!inflater.finished()) {
-                throw new SAMLException("Out of space allocated for inflated data");
+       try{
+           byte[] out = data;
+           int length = data.length;
+            try {
+                Inflater inflater = new Inflater(true);
+                inflater.setInput(data);
+                byte[] xmlMessageBytes = new byte[MAX_INFLATED_SIZE];
+                int inflatedLength = inflater.inflate(xmlMessageBytes);
+                if (!inflater.finished()) {
+                    throw new SAMLException("Out of space allocated for inflated data ("+(MAX_INFLATED_SIZE/1000)+"kb)");
+                }
+                inflater.end();
+                out = xmlMessageBytes;
+                length = inflatedLength;
+            } catch (DataFormatException e) {
+                log.warn("Inflate SAML message failed - Invalid data format", e);
```

Comment:
I'd suggest debug since it's now expected to not be compressed (otherwise it just logs a lot of unnecessary warns).